src/xdppcw/inc/xdppcw.h

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+#pragma once
+
+#define MAXUINT32_STR "4294967295"
+
+typedef struct _PCW_INSTANCE PCW_INSTANCE;
+
+typedef struct _XDP_PCW_RX_QUEUE {
+    UINT64 XskFramesDelivered;
+    UINT64 XskFramesDropped;
+    UINT64 XskFramesTruncated;
+    UINT64 XskInvalidDescriptors;
+    UINT64 InspectBatches;
+    UINT64 InspectFramesPassed;
+    UINT64 InspectFramesDropped;
+    UINT64 InspectFramesRedirected;
+    UINT64 InspectFramesForwarded;
+} XDP_PCW_RX_QUEUE;
+
+typedef struct _XDP_PCW_LWF_RX_QUEUE {
+    UINT64 MappingFailures;
+    UINT64 LinearizationFailures;
+    UINT64 ForwardingFailures;
+} XDP_PCW_LWF_RX_QUEUE;
+
+typedef struct _XDP_PCW_TX_QUEUE {
+    UINT64 XskInvalidDescriptors;
+    UINT64 InjectionBatches;
+    UINT64 QueueDepth;
+} XDP_PCW_TX_QUEUE;
+
+typedef struct _XDP_PCW_LWF_TX_QUEUE {
+    UINT64 FramesDroppedPause;
+    UINT64 FramesDroppedNic;
+} XDP_PCW_LWF_TX_QUEUE;
+
+#define STAT_INC(_Stats, _Field) (((_Stats)->_Field)++)
+#define STAT_ADD(_Stats, _Field, _Bias) (((_Stats)->_Field) += (_Bias))
+#define STAT_SET(_Stats, _Field, _Value) (((_Stats)->_Field) = (_Value))
```

Comment:
Are these not interlocked because you don't expect them to contend?

Reply:
Right, these are not interlocked by design. If we add counters that may be contended, we'd add new helpers.